// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";
option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.segmentation_platform.proto";
option java_outer_classname = "PredictionResultProto";

package segmentation_platform.proto;

import "components/segmentation_platform/public/proto/output_config.proto";

// Result from the model evaluation for a given segment.
message PredictionResult {
  // The result is the confidence rating from the model evaluation.
  // This field will be deprecated going forward.
  repeated float result = 1;

  // The time when the prediction was made, in terms of the number of
  // microseconds since Windows epoch.
  optional int64 timestamp_us = 2;

  // The output config from the metadata.
  optional OutputConfig output_config = 3;
}
